### PR TITLE
Add experimental_aggregate_gradients support

### PR DESCRIPTION
**Pull request by bhack**
Try to fix https://github.com/tensorflow/addons/issues/1920

### CODE REVIEW

**Comment by bhack** [comment]
Let me know if it makes sense to add this test:

```
def test_fit_simple_linear_model_mixed_precision():
    np.random.seed(0x2019)
    tf.random.set_seed(0x2019)

    x = np.random.standard_normal((10000, 3))
    w = np.random.standard_normal((3, 1))
    y = np.dot(x, w) + np.random.standard_normal((10000, 1)) * 1e-4

    tf.keras.mixed_precision.experimental.set_policy("mixed_float16")
    model = tf.keras.models.Sequential()
    model.add(tf.keras.layers.Dense(input_shape=(3,), units=1))
    model.compile(Lookahead("sgd"), loss="mse")

    model.fit(x, y, epochs=3)

    x = np.random.standard_normal((100, 3))
    y = np.dot(x, w)
    predicted = model.predict(x)

    max_abs_diff = np.max(np.abs(predicted - y))
    assert max_abs_diff < 2.3e-3
```

File: tensorflow_addons/optimizers/tests/lookahead_test.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import tensorflow as tf
 
 from tensorflow_addons.optimizers import Lookahead
+from tensorflow.python.keras.mixed_precision.experimental import policy
```
Comment (bhack) [inline tensorflow_addons/optimizers/tests/lookahead_test.py:22]: One of the check is failing cause our policy doesn't let us to import this namespace but `policy_scope` is not exposed and in many parts there are `thread_local`  todos like in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/mixed_precision/experimental/policy.py#L484

Comment (bhack) [reply tensorflow_addons/optimizers/tests/lookahead_test.py:22]: See also https://github.com/tensorflow/tensorflow/pull/40475

Comment (seanpmorgan) [reply tensorflow_addons/optimizers/tests/lookahead_test.py:22]: Seeing as we have no public API to make this thread local, I'm in favor of including this test case and adding the file to the list of allowed private usage:
https://github.com/tensorflow/addons/blob/master/tools/testing/source_code_test.py#L86

This has been confirmed by the user to solve the problem, and the private usage is not called in the optimizer source code, only the test case. Seems counterproductive to not include the test. If https://github.com/tensorflow/tensorflow/pull/40475 or some variant merges we can remove it. @gabrieldemarmiesse thoughts?

File: tensorflow_addons/optimizers/tests/lookahead_test.py [hunk 2]
```diff
@@ -120,6 +121,30 @@ def test_fit_simple_linear_model():
...
+
+    model.fit(x, y, epochs=3)
+
+    x = np.random.standard_normal((100, 3))
+    y = np.dot(x, w)
+    predicted = model.predict(x)
+
+    max_abs_diff = np.max(np.abs(predicted - y))
+    assert max_abs_diff < 2.3e-3
+    assert max_abs_diff >= 1e-3
```
Comment (seanpmorgan) [inline tensorflow_addons/optimizers/tests/lookahead_test.py:145]: Why assert a greater than difference here? Mixed precision variation?

Comment (bhack) [reply tensorflow_addons/optimizers/tests/lookahead_test.py:145]: It Is what I think. There Is a default dynamic loss scaling to avoid numerical precision underrun https://www.tensorflow.org/api_docs/python/tf/keras/mixed_precision/experimental/Policy?version=nightly
